libgadget/tests/test_hci.c

```diff
@@ -156,16 +156,16 @@ test_hci_terminate(void ** state)
     assert_int_equal(action->write_snapshot, 0);
 }
 
-static void setup(void ** state)
+static int setup(void ** state)
 {
-    mkdtemp(prefix);
+    char * ret = mkdtemp(prefix);
     message(0, "UsingPrefix : '%s'\n", prefix);
-
+    return !ret;
 }
 
-static void teardown(void ** state)
+static int teardown(void ** state)
 {
-    remove(prefix);
+    return remove(prefix);
```

Comment:
likely non-resursive and will fail.

Reply:
You mean that the remove call will fail if the directory is non-empty? I think that's ok - the subfiles should be removed by the HCI manager, shouldn't they? This checks that they are.

Reply:
Sure. But that means it is a test. I think it is non-critical to remove or not. Sometimes people do `system("rm -rf")` for the clean up.

Reply:
Fair enough, I can change it to remove(prefix); return 0;